Inkam revisi

raihaninkam opened the pull request:
## Revision Summary  
This PR addresses shortcomings in previous commits by:  

### 🔍 Issues in Previous Commits  
 **Overly Brief Commit Messages**  
   - Only contained generic descriptions (e.g., `"feat: create central module hub"`)  
   - Lacked details about specific changes  



###  Improvements Made  
 **Commit Message Enhancements**  
   - Added detailed descriptions for each function:  
     ```bash
     git commit -m "feat(triangle): implement number pattern with validation
     - Validates for positive numbers only
     - Output: ascending pattern (1\n1 2\n...)"
     ```

raihaninkam committed:
feat(task-1): implement triangle number pattern

- Generates ascending number triangle
- Validates for positive number input

raihaninkam committed:
feat(task-2): create array product calculator

- Multiplies all array elements
- Handles non-array inputs
- Example: countArray([2,3,4]) returns 24

raihaninkam committed:
feat(task-1): implement number triangle generator

- Creates ascending number pattern using nested loops
- Validates for positive integer input
- Example: triangle(3) → '1\n1 2\n1 2 3'

raihaninkam committed:
feat(task-2): add array product calculator

- Multiplies all elements in array
- Returns 1 for empty arrays
- Handles non-array inputs with error message

raihaninkam committed:
feat(task-5): create multiplication table function

- Generates 10x multiplication table
- Validates numeric input
- Example: cetakTabel(3) prints table up to 3x10

raihaninkam committed:
feat(task-6): implement range generator

- Outputs numbers from start to end (inclusive)
- Handles reverse ranges
- Example: buatRentang(3,5) → '3, 4, 5'

raihaninkam committed:
docs: add error documentation file

- Catalogues 5 common runtime errors
- Includes file locations and line numbers
- Documents root causes and solutions
- Covers:
  • ReferenceError (user not defined)
  • TypeError (null.length)
  • SyntaxError (unexpected token)
  • RangeError (call stack exceeded)

raihaninkam committed:
feat(index): create central module hub

- Aggregate all task functions (1.js-7.js)

kodacampmain commented:
Berikutnya di lanjut dari yang lama aja ya, jadi branch yang tadi diedit langsung, tidak usah membuat branch baru

kodacampmain commented:
file index.js yang dimaksud masih belum tepat, diperhatikan lagi readme dan contoh kode yang sudah ada @raihaninkam

raihaninkam committed:
feat: export function

raihaninkam committed:
feat(task-1): implement number triangle generator

- Creates ascending number pattern using nested loops
- Validates for positive integer input
- Example: triangle(3) → '1\n1 2\n1 2 3'

raihaninkam committed:
feat(task-2): add array product calculator

- Multiplies all elements in array
- Returns 1 for empty arrays
- Handles non-array inputs with error message
- Test case: countArray([2,3]) → 6

raihaninkam committed:
feat(task-2): add array product calculator

- Multiplies all elements in array
- Returns 1 for empty arrays
- Handles non-array inputs with error message
- Test case: countArray([2,3]) → 6

raihaninkam committed:
feat(task-5): create multiplication table function

- Generates 10x multiplication table
- Validates numeric input
- Example: cetakTabel(3) prints table up to 3x10

raihaninkam committed:
feat(task-6): implement range generator

- Outputs numbers from start to end (inclusive)
- Handles reverse ranges
- Example: buatRentang(3,5) → '3, 4, 5'

raihaninkam committed:
feat: add error docs

kodacampmain commented:
tugas diletakkan di direktori yang sudah ditentukan di readme @raihaninkam

kodacampmain commented:
import di index.js direktori nya tidak sesuai dengan yang ada
import dari /tasks/inkam
folder pengerjaan ada di /inkam

raihaninkam committed:
feat(index.js): add module.exports to aggregate all utility functions

raihaninkam committed:
feat(index.js): import inkam module from tasks/inkam
 implement number triangle generator:
- Creates ascending number pattern using nested loops
- Validates for positive integer input
- Example: triangle(3) → '1\n1 2\n1 2 3'

raihaninkam committed:
feat(task-2): add array product calculator

- Multiplies all elements in array
- Returns 1 for empty arrays
- Handles non-array inputs with error message

raihaninkam committed:
feat(task-1): implement number triangle generator

- Creates ascending number pattern using nested loops
- Validates for positive integer input

raihaninkam committed:
feat: add array product calculator

- Multiplies all elements in array
- Returns 1 for empty arrays
- Handles non-array inputs with error message
- Test case: countArray([2,3]) → 6

raihaninkam committed:
feat(task-5): create multiplication table function

- Generates 10x multiplication table
- Validates numeric input
- Example: cetakTabel(3) prints table up to 3x10

raihaninkam committed:
feat: implement range generator

- Outputs numbers from start to end (inclusive)
- Handles reverse ranges
- Example: buatRentang(3,5) → '3, 4, 5'

raihaninkam committed:
feat: function call

raihaninkam committed:
fix: resolve error in getUserData

- Initialize 'user' variable before access
- Add null checks for user object
- Update calling functions (processUser/main)
- Related to: getUserData() at app.js:15

raihaninkam committed:
resolve the conflict

kodacampmain commented:
diperhatikan lagi cara pengerjaannya di readme

raihaninkam committed:
feat: implement number triangle generator

- Creates ascending number pattern using nested loops
- Validates for positive integer input
- Example: triangle(3) → '1\n1 2\n1 2 3'

kodacampmain commented:
gunakan try-catch kalau ada error yang bisa ditangkap
dilihat lagi error yang kamu berikan apakah bisa ditangkap oleh catch
perhatikan juga bagaimana membuat error bisa ditangkap oleh catch

raihaninkam committed:
feat:

- Multiplies all elements in array
- Returns 1 for empty arrays
- Handles non-array inputs with error message
- Test case: countArray([2,3]) → 6

raihaninkam committed:
feat: implement reverse triangle generator
>>
>> - Creates ascending number pattern using nested loops
>> - Validates for positive integer input

raihaninkam committed:
feat:

- Filters and multiplies odd numbers
- Returns 1 if no odds found
- Example: processNumbers([1,2,3]) → 3

raihaninkam committed:
feat:

- Generates 10x multiplication table
- Validates numeric input
- Example: cetakTabel(3) prints table up to 3x10

raihaninkam committed:
feat:

- Generates 10x multiplication table
- Validates numeric input
- Example: cetakTabel(3) prints table up to 3x10

raihaninkam committed:
feat(task-1): implement number triangle generator

- Creates ascending number pattern using nested loops
- Validates for positive integer input
- Example: triangle(3) → '1\n1 2\n1 2 3'

raihaninkam committed:
feat(task-2): add array calculator

- Multiplies all elements in array
- Returns 1 for empty arrays
- Handles non-array inputs with error message
- Test case: countArray([2,3]) → 6

raihaninkam committed:
feat(task-3): implement number reverse triangle generator

- Creates ascending number pattern using nested loops
- Validates for positive integer input
- Example: triangle(3) → '1\n1 2\n1 2 3'

raihaninkam committed:
feat(task-7): add odd number processor

- Filters and multiplies odd numbers
- Returns 1 if no odds found

raihaninkam committed:
feat(task-5): create multiplication table function

- Generates 10x multiplication table
- Validates numeric input
- Example: cetakTabel(3) prints table up to 3x10

raihaninkam committed:
feat(task-6): implement range generator

- Outputs numbers from start to end (inclusive)
- Handles reverse ranges
- Example: buatRentang(3,5) → '3, 4, 5'

raihaninkam committed:
feat: Error Documentation

raihaninkam committed:
feat(index.js): import inkam module from tasks/inkam
and funtion call

raihaninkam committed:
resolve conflict

raihaninkam committed:
fix: Delete folder inkam in root

raihaninkam committed:
fix: delete double 'taskInkam' function

raihaninkam committed:
fix: fixed try catch handle on file index.js

raihaninkam committed:
fix: add throw new error to fixed try catch error handling

raihaninkam committed:
fix: double or missing throw on module

raihaninkam committed:
fix: add 'throw' on error handling

raihaninkam committed:
fix: add 'throw' to error handling module

raihaninkam committed:
fix: add 'throw for error handling in module

raihaninkam committed:
fix: add 'throw' for error handling

raihaninkam committed:
fix: add 'throw' for error handling

kodacampmain commented:
done
